Patch taken from Debian: revert changes that prevent building against freerdp
version 1.1 branch.

From 8ebc0685b85e0d1f70eb00171f2e7712de3d44bd Mon Sep 17 00:00:00 2001
From: Michael Biebl <biebl@debian.org>
Date: Thu, 22 Sep 2016 01:15:55 +0200
Subject: [PATCH 1/2] Revert "Improve FreeRDP authentication failure handling"

This reverts commit d7b4f88943e8615d252d27e1efc58cb64a9e1821.
---
 plugins/rdp/vinagre-rdp-tab.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/plugins/rdp/vinagre-rdp-tab.c b/plugins/rdp/vinagre-rdp-tab.c
index b731f9b..8572bc3 100644
--- a/plugins/rdp/vinagre-rdp-tab.c
+++ b/plugins/rdp/vinagre-rdp-tab.c
@@ -1195,8 +1195,8 @@ open_freerdp (VinagreRdpTab *rdp_tab)
   VinagreTab           *tab = VINAGRE_TAB (rdp_tab);
   GtkWindow            *window = GTK_WINDOW (vinagre_tab_get_window (tab));
   gboolean              success = TRUE;
+  gboolean              authentication_error = FALSE;
   gboolean              cancelled = FALSE;
-  guint                 authentication_errors = 0;
 
   priv->events = g_queue_new ();
 
@@ -1205,12 +1205,14 @@ open_freerdp (VinagreRdpTab *rdp_tab)
 
   do
     {
+      authentication_error = FALSE;
+
       /* Run FreeRDP session */
       success = freerdp_connect (priv->freerdp_session);
       if (!success)
         {
-          authentication_errors += freerdp_get_last_error (priv->freerdp_session->context) == 0x20009 ||
-                                   freerdp_get_last_error (priv->freerdp_session->context) == 0x2000c;
+          authentication_error = freerdp_get_last_error (priv->freerdp_session->context) == 0x20009 ||
+                                 freerdp_get_last_error (priv->freerdp_session->context) == 0x2000c;
 
           cancelled = freerdp_get_last_error (priv->freerdp_session->context) == 0x2000b;
 
@@ -1218,7 +1220,7 @@ open_freerdp (VinagreRdpTab *rdp_tab)
           init_freerdp (rdp_tab);
         }
     }
-  while (!success && authentication_errors < 3);
+  while (!success && authentication_error);
 
   if (!success)
     {
-- 
2.9.3

